src/components/ActionList/README.md

```diff
@@ -85,49 +85,41 @@ Each item in an action list should be scannable avoiding unnecessary words and a
 Use for the least important actions so merchants aren’t distracted by secondary tasks. Can also be used for a set of actions that won’t fit in the available screen space.
 
 ```jsx
-class ActionListExample extends React.Component {
-  state = {
-    active: true,
-  };
-
-  togglePopover = () => {
-    this.setState(({active}) => {
-      return {active: !active};
-    });
-  };
-
-  render() {
-    const activator = (
-      <Button onClick={this.togglePopover}>More actions</Button>
-    );
-
-    return (
-      <div style={{height: '250px'}}>
-        <Popover
-          active={this.state.active}
-          activator={activator}
-          onClose={this.togglePopover}
-        >
-          <ActionList
-            items={[
-              {
-                content: 'Import file',
-                onAction: () => {
-                  console.log('File imported');
-                },
-              },
-              {
-                content: 'Export file',
-                onAction: () => {
-                  console.log('File exported');
-                },
-              },
-            ]}
-          />
-        </Popover>
-      </div>
-    );
-  }
+function ActionListExample() {
+  const [active, setActive] = useState(true);
+
+  const toggleActive = useCallback(() => {
+    setActive(!active);
+  }, [active]);
+
+  const handleImportedAction = useCallback(() => {
+    console.log('Imported action');
+  }, []);
```

Comment:
Is this code people might be copy-pasting into Shopify/web and then adapting to their needs? If so, the perf optimization may be worth it?

Reply:
1 - It's premature optimization
2 - It might not be an optimization at all

The belief that arrow functions are slow is unfounded, and using them blindly like this along with PureComponents can actually make the code slower. Me and Ben made an issue to address this: https://github.com/Shopify/polaris-react/issues/2062

Reply:
Thank you for the link, great issue!

Reply:
I've gone back and forth on this and I'm leaning towards "lets just always use useCallback".

Yes it's a bit more noisy in the cases where you're creating a function that will be passed into a non-pure component but it's one less decision to make when coding compared to "go check if you're passing this into a PureComponent, if you are then use useCallback otherwise don't". It will also help insulate us against future changes when we decide we need to make a regular component be PureComponent or wrapped in React.memo -because all call sites should already using useCallback they're PureComponent-friendly, so we won't have to update all the call sites to not break the optimization as they're already playing nice.

And following on from that we should write our examples in the same way we'd write our code so our examples should use useCallback too.